### PR TITLE
pythonPackages.onnx: init at version 1.6.0

### PR DESCRIPTION
**Pull request by acairncross**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jonringer** [comment]
oh lol, i was thinking of putting this up as well

File: pkgs/development/python-modules/onnx/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    patchShebangs tools/protoc-gen-mypy.py
+  '';
+
+  # The setup.py does all the configuration (running CMake)
+  dontConfigure = true;
+
+  meta = {
+    homepage    = http://onnx.ai;
+    description = "Open Neural Network Exchange";
+    license     = lib.licenses.mit;
```
Comment (jonringer) [inline pkgs/development/python-modules/onnx/default.nix]: do you mind adding yourself as a maintainer?

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.onnx python37Packages.onnx

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
leaf package

```
[4 built, 1 copied (0.2 MiB), 0.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71202
2 package were build:
python27Packages.hcloud python37Packages.hcloud
```

only thing is that it seems to install some test utilties which I don't think are too important:
```
[nix-shell:/home/jon/.cache/nix-review/pr-71211]$ tree ./results/python27Packages.onnx/bin/
./results/python27Packages.onnx/bin/
├── backend-test-tools
├── check-model
└── check-node
```
you can remove these with:
```
  postInstall = ''
    rm -r $out/bin
  '';
```

File: pkgs/development/python-modules/onnx/default.nix (hunk 1 above)
Comment (acairncross) [reply pkgs/development/python-modules/onnx/default.nix]: Done

**Comment by acairncross** [comment]
Woops, didn't really mean to re-request review!

I've removed the installed executables.

**Comment by jonringer** [comment]
python2 version seems to be broken

```
[nix-shell:/home/jon/.cache/nix-review/pr-71211-1]$ nix-shell --pure -p "with import ./nixpkgs {}; pythonPackages.onnx"

[nix-shell:/home/jon/.cache/nix-review/pr-71211-1]$ python
Python 2.7.16 (default, Mar  2 2019, 18:34:01)
[GCC 8.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import onnx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/p5257ap9szl0x8zsn5g920b4nrcnviq2-python2.7-onnx-1.6.0/lib/python2.7/site-packages/onnx/__init__.py", line 9, in <module>
    from onnx.external_data_helper import load_external_data_for_model, write_external_data_tensors
  File "/nix/store/p5257ap9szl0x8zsn5g920b4nrcnviq2-python2.7-onnx-1.6.0/lib/python2.7/site-packages/onnx/external_data_helper.py", line 10, in <module>
    from .onnx_pb import TensorProto, ModelProto
  File "/nix/store/p5257ap9szl0x8zsn5g920b4nrcnviq2-python2.7-onnx-1.6.0/lib/python2.7/site-packages/onnx/onnx_pb.py", line 8, in <module>
    from .onnx_ml_pb2 import *  # noqa
  File "/nix/store/p5257ap9szl0x8zsn5g920b4nrcnviq2-python2.7-onnx-1.6.0/lib/python2.7/site-packages/onnx/onnx_ml_pb2.py", line 7, in <module>
    from google.protobuf.internal import enum_type_wrapper
ImportError: No module named google.protobuf.internal
```

**Comment by acairncross** [comment]
Seems to be a problem with the protobuf installation. In particular because Python 2 doesn't support [implicit namespace packages](https://www.python.org/dev/peps/pep-0420/). Chucking `__init__.py`s into the `google` and `google/protobuf` directories of the installation seems to "fix" the problem, although I doubt that's the proper/clean solution (that's certainly not what pip does). What I don't understand is why the pip-installed version doesn't have the same issue. Also I wonder why this hasn't been a (reported) issue for any other Python packages using protobuf.

**Comment by jonringer** [comment]
pip installed version likely dumps it into the same directory as all the other packages, so it gets to fuse all directory paths into one coherent path. Nixpkgs doesn't get this benefit since it's a series of storepaths.

Seeing as python2 is EOL in 1.5 months, I'm fine with just saying `disabled = isPy27` and moving on. @FRidh thoughts?

File: pkgs/development/python-modules/onnx/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    sha256 = "0ig33jl3591041lyylxp52yi20rfrcqx3i030hd6al8iabzc721v";
+  };
+
+  # Remove the unqualified requirement for the typing package for running the
+  # tests. typing is already required for the installation, where it is
+  # correctly qualified so as to only be required for sufficiently old Python
+  # versions.
+  # This patch has been merged upstream so can be removed upon the next version
+  # bump.
+  patches = [ ./no-typing.patch ];
```
Comment (Taneb) [inline pkgs/development/python-modules/onnx/default.nix]: You should use `fetchpatch` from the upstream for patches that aren't nixpkgs-specific.

**Comment by jonringer** [comment]
please squash the commits:

```
git reset HEAD^
git add pkgs/
git commit --amend --no-edit
git push <fork> <branch> --force
```

**Comment by acairncross** [comment]
Done

File: pkgs/development/python-modules/onnx/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
+{ lib
+, fetchpatch
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
```
Comment (jonringer) [inline pkgs/development/python-modules/onnx/default.nix]:
```suggestion
, pythonOlder
, isPy27
```

**Review by jonringer — COMMENTED** [review]
python2 build fails due to namespace import errors on protobuf

**Review by jonringer — APPROVED** [review]
LGTM

```
[1 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71211
2 package were built:
python37Packages.onnx python38Packages.onnx
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.onnx python38Packages.onnx

**Pull request merged**